src/dc_chat.rs

```diff
@@ -1423,19 +1423,11 @@ pub fn dc_get_chat_contacts(context: &Context, chat_id: u32) -> *mut dc_array_t
             "SELECT cc.contact_id FROM chats_contacts cc \
              LEFT JOIN contacts c ON c.id=cc.contact_id WHERE cc.chat_id=? \
              ORDER BY c.id=1, LOWER(c.name||c.addr), c.id;",
-            params![chat_id as i32],
-            |row| row.get::<_, i32>(0),
-            |ids| {
-                let mut ret = dc_array_t::new(100);
-
-                for id in ids {
-                    ret.add_id(id? as u32);
-                }
-
-                Ok(ret.into_raw())
-            },
+            params![chat_id],
+            |row| row.get::<_, u32>(0),
+            |ids| Ok(ids.collect::<Result<Vec<_>, _>>()?),
```

Comment:
Can't get rid of this `Ok` and `?` because of
```
expected enum `error::Error`, found enum `rusqlite::Error`
```

Reply:
`ids.collect::<Result<Vec<_>, _>>().map_err(Into::into)` will do the trick

Reply:
Fixed, thanks